x/batching/keeper/keeper.go

```diff
@@ -72,6 +74,39 @@ func NewKeeper(
 	return k
 }
 
+// SetDataResultForBatching stores a data result so that it is ready
+// to be batched.
+func (k Keeper) SetDataResultForBatching(ctx context.Context, result types.DataResult) error {
+	return k.dataResults.Set(ctx, collections.Join(false, result.DrId), result)
+}
+
+// setDataResultAfterBatching stores a data result after batching
+// under the status "batched."
+func (k Keeper) setDataResultAfterBatching(ctx context.Context, result types.DataResult) error {
+	return k.dataResults.Set(ctx, collections.Join(true, result.DrId), result)
+}
```

Comment:
Won't this leave the 'unbatched' data results in the store?

Reply:
Maybe the second method can be something like `markResultAsBatched` and it takes care of removing the unbatched variant and storing the batched result?

Reply:
Ah good catch. Thank you.